components/email_aliases/email_aliases.mojom

```diff
@@ -0,0 +1,62 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+module email_aliases.mojom;
+
+struct Alias {
+  string email;
+  string? note;
+  array<string>? domains;
+};
+
+// Observer for change in email aliases state
+interface EmailAliasesServiceObserver {
+
+  // Called when aliases have been updated.
+  OnAliasesUpdated(array<Alias> aliases);
```

Comment:
how do we get the list of aliases that we already have without changing them?

Reply:
Same as OnLoggedIn (see comment above)

Reply:
This one was actually my suggestion @bridiver. We do it in a few other places (like Brave News) and its just a bit of a convenience thing:

```ts
// instead of 
api.addListener(newState => setState(newState))
setState(await api.getState())

// you can do
api.addListener(newState => setState(newState))
// and be notified of the initial state.
```